maintenance/shaka-player.md

```diff
@@ -0,0 +1,185 @@
+# Shaka Player Maintenance Process
+
+These are project-specific processes in addition to the
+[general release process documentation](./).
+
+
+## Release Branches
+
+Feature release are made from the `main` branch.  Any time that happens, our
+release automation will automatically create a release branch.  So, for
+example, when `v4.1.0` was released from `main`, a branch named `v4.1.x` was
+automatically created.
+
+Future bug fixes would be cherry-picked from `main` to `v4.1.x` to create the
+releases `v4.1.1`, `v4.1.2`, etc., and the branch would become more stable over
+time.
+
+
+## Cherry-Picking
+
+Features are never cherry-picked to a release branch.  This would go against
+[semantic versioning](https://semver.org/), which says that a patch-level
+release (third digit) can only contain backward-compatible bug fixes.
+Features scoped to test infrastructure or the demo app are an exception, since
+those are not part of the library itself.
+
+### Interactive Cherry-Pick Tool
+
+We use a [custom tool](https://github.com/shaka-project/shaka-github-tools/tree/main/interactive-cherry-pick)
+to perform an interactive cherry-pick.  This is similar to the interactive
+rebase of `git`.  When you run the tool, your default editor opens with a list
+of commits you may want to cherry-pick.  Any commits left in the file when you
+quit the editor are cherry-picked using `git`.
+
+If a commit fails to cherry-pick, the maintainer is given an error message from
+git and must either fix the conflict or instruct git to skip the commit.
+
+Cherry-picks can be done 1 or more times between bug-fix releases.  The release
+is finally made by merging the release PR for the branch.
+
+For example, after installing the tool:
+
+```sh
+# Go to the Shaka Player repo.
+cd /path/to/shaka-player
+
+# Check out a release branch tracking the _upstream_ repo:
+git checkout -b v3.3.x upstream/v3.3.x
+# Or if you already have such a branch locally:
+git checkout v3.3.x
+
+# Inspect all commits since a particular main-branch tag.
+# If the most recent v3.3.x release was v3.3.1, the tag
+# v3.3.1-main is the same point in history for the main branch.
+interactive-cherry-pick v3.3.1-main upstream main
```

Comment:
Here I have discovered that only the tag is mandatory, the rest of the parameters have a default value, should it be mentioned here?

Reply:
That's true if you set up your local clone the way we recommend.  I wanted to mention the parameters, though, so it's clear that they exist and that you might need to set them.

I'll see if I can find a concise way to explain all that in the doc.

Reply:
Done